src/server/frames.ts

```diff
@@ -1371,3 +1373,15 @@ function verifyLifecycle(name: string, waitUntil: types.LifecycleEvent): types.L
     throw new Error(`${name}: expected one of (load|domcontentloaded|networkidle)`);
   return waitUntil;
 }
+
+function constructNavigationUrl(baseUrl: string | undefined, givenUrl: string): string {
+  if (!baseUrl)
+    return givenUrl;
+  try {
+    new URL(givenUrl);
+    return givenUrl;
+  } catch (error) {
+    const optionalSlash = (givenUrl && baseUrl[baseUrl.length] !== '/' && givenUrl[0] !== '/') ? '/' : '';
```

Comment:
- `baseUrl[baseUrl.length]` is always undefined
- Let's make `http://localhost:3000/build/` and `/foo.html` play together nicely?